client/db/src/lib.rs

```diff
@@ -672,8 +681,16 @@ impl<Block: BlockT> sc_state_db::NodeDb for StorageDb<Block> {
 	type Error = io::Error;
 	type Key = [u8];
 
-	fn get(&self, key: &[u8]) -> Result<Option<Vec<u8>>, Self::Error> {
-		self.db.get(columns::STATE, key).map(|r| r.map(|v| v.to_vec()))
+	fn get(&self, child_info: &ChildInfo, key: &[u8]) -> Result<Option<Vec<u8>>, Self::Error> {
+		if child_info.is_top_trie() {
+			self.db.get(columns::STATE, key)
+		} else {
+			let keyspace = child_info.keyspace();
+			let mut key_buffer = vec![0; keyspace.len() + key.len()];
```

Comment:
Can we not reuse this `key_buffer`?

Reply:
I tried to a bit, but this means inner mutability, do you think it is worth it?
(I also did try to switch the primitive to get(&mut self... but this was a bit to impacting, I could try a bit more).